File: 2966-divide-array-into-arrays-with-max-difference/2966-divide-array-into-arrays-with-max-difference.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[][]}
 */

var divideArray = function(nums, k) {

    check = (answer) => {
        for (row of answer){
            if (row[row.length - 1] - row[0] > k){
                return false;
            }
        }

        return true;
    } 

    if (nums.length < 3 || nums.length % 3 != 0){
        return [];
    }

    nums.sort((a, b) => a - b);
    let answer = [];
    for (let i = 0; i < nums.length; i += 3){
        answer.push(nums.slice(i, i + 3));
    }
    return check(answer) ? answer : [];
};

File: 2966-divide-array-into-arrays-with-max-difference/2966-divide-array-into-arrays-with-max-difference.py
class Solution:
    def divideArray(self, nums: List[int], k: int) -> List[List[int]]:
        def check():
            for arr in answer:
                if arr[-1] - arr[0] > k:
                    return False
                
            return True

        if len(nums) < 3 or len(nums) % 3:
            return []
            
        nums.sort()
        answer = []
        for i in range(0, len(nums), 3):
            answer.append(nums[i:i + 3])
        
        return answer if check() else []
 